noobaa/noobaa-core
# block_store_fs enhanced to use xattrs and migrate blocks to tier2

### Explain the changes
1. Enhance backing_store_fs to use xattr to store block_md json instead of in separate .meta files.
2. This allows reducing the number of files needed on the file system by half.
3. Add new config option (false by default) - `config.BLOCK_STORE_FS_TIER2_ENABLED` - enable to set custom xattr trigger to signal the filesystem to migrate the data to archive tier (for filesystems that support that).
4. Added new config option (false by default) - `config.BLOCK_STORE_FS_MAPPING_INFO_ENABLED ` - enable to pass on extra recovery information (which bucket, object, offset, etc) to be added in block_md xattr, for cases when recovery without the db is needed.

### Issues: Fixed #xxx / Gap #xxx
1. NA

### Testing Instructions:
1. Regression of backing_store_fs (PV pools from the operator).
2. Enable the new configs to test the new options and check the xattr of the blocks (e.g `xattr -l <filename>` on mac).
